docs/book/developer-guide/repo-and-config.md

```diff
@@ -2,8 +2,309 @@
+`zenml.repository.Repository` singleton acts as the central point of
+management for Stacks, Stack Components, Stack Component Flavorsand other
+associated ZenML concepts. The following are examples of how to use these
+singletons to perform various operations:
+
+* to retrieve the global configuration, active profile and active stack:
+
+    ```python
+    from zenml.config.global_config import GlobalConfiguration
+    from zenml.repository import Repository
+
+    repo = Repository()
+    config = GlobalConfiguration()
+
+    # the unique user UUID set
+    user_id = config.user_id
+
+    # the active profile name
+    profile = repo.active_profile_name
+
+    # the active profile object
+    profile = repo.active_profile
+
+    # the name of the active stack
+    active_stack_name = config.active_stack_name
+
+    # the active stack object
+    active_stack = repo.active_stack
+    ```
+
+* to create a new Profile and set it as the active one:
+
+    ```python
+    from zenml.repository import Repository
+    from zenml.config.global_config import GlobalConfiguration
+    from zenml.config.profile_config import ProfileConfiguration
+
+    repo = Repository()
+    config = GlobalConfiguration()
+
+    profile = ProfileConfiguration(name="local")
+    config.add_or_update_profile(profile)
+
+    repo.activate_profile("local")
+    ```
+
+* access the active stack
+
+    ```python
+    from zenml.repository import Repository
+
+    repo = Repository()
+    stack = repo.active_stack
+    
+    print(stack.name)
+    print(stack.orchestrator.name)
+    print(stack.artifact_store.name)
+    print(stack.artifact_store.path)
+    print(stack.metadata_store.name)
+    print(stack.metadata_store.uri)
+    ```
+
+* register and set a new stack
+
+    ```python
+    from zenml.repository import Repository
+    from zenml.artifact_stores import LocalArtifactStore
+    from zenml.metadata_stores import SQLiteMetadataStore
+    from zenml.orchestrators import LocalOrchestrator
+    from zenml.stack import Stack
+
+    repo = Repository()
+
+    orchestrator = LocalOrchestrator(name="local")
+    metadata_store = SQLiteMetadataStore(
+        name="local",
+        uri="/tmp/zenml/zenml.db",
+    )
+    artifact_store = LocalArtifactStore(
+        name="local",
+        path="/tmp/zenml/artifacts",
+    )
+    stack = Stack(
+        name="local",
+        orchestrator=orchestrator,
+        metadata_store=metadata_store,
+        artifact_store=artifact_store,
+    )
+
+    repo.register_stack(stack)
+    repo.activate_stack(stack.name)
+    ```
+
+To explore all possible operations that can be performed via the `GlobalConfiguration`
+and `Repository` singletons, please consult the API docs sections on [GlobalConfiguration](https://apidocs.zenml.io/0.7.3/api_docs/config/#zenml.config.global_config.GlobalConfiguration) and [Repository](https://apidocs.zenml.io/0.7.3/api_docs/repository/#zenml.repository.Repository).
```

Comment:
Do we want to have this set to `latest` or something like that so people aren't browsing outdated API docs?

Reply:
Nice catch ! thanks !